src/main/java/org/digitalstack/chess/ChessBoard.java

```diff
@@ -12,10 +12,36 @@ public ChessBoard() {
     }
 
     public void add(Pawn pawn, int xCoordinate, int yCoordinate, PieceColor pieceColor) {
-        throw new UnsupportedOperationException("Need to implement ChessBoard.add()");
+        int numberOfWhitePawns = 0;
+        int numberOfBlackPawns = 0;
+
+        for (Pawn[] col : pieces) {
+            for (Pawn row : col) {
+                if (row != null) {
+                    if (pieceColor.equals(PieceColor.WHITE)) numberOfWhitePawns += 1;
+                    else numberOfBlackPawns += 1;
+                }
+            }
+        }
+
+        if (!validNrOfPawns(pieceColor, numberOfWhitePawns, numberOfBlackPawns)
+        || !isLegalBoardPosition(pawn.getXCoordinate(), pawn.getYCoordinate())
+        || pieces[xCoordinate][yCoordinate] != null) {
+            pawn.setXCoordinate(-1);
+            pawn.setYCoordinate(-1);
+
+        } else if (pieces[xCoordinate][yCoordinate] == null) {
+            pawn.setXCoordinate(xCoordinate);
+            pawn.setYCoordinate(yCoordinate);
+            pieces[pawn.getXCoordinate()][pawn.getYCoordinate()] = pawn;
+        }
+    }
+
+    private boolean validNrOfPawns(PieceColor pieceColor, int numberOfWhitePawns, int numberOfBlackPawns) {
+        return PieceColor.WHITE.equals(pieceColor) && numberOfWhitePawns < 7 || PieceColor.BLACK.equals(pieceColor) && numberOfBlackPawns < 7;
```

Comment:
Shouldn't this be:
(PieceColor.WHITE.equals(pieceColor) && numberOfWhitePawns < 7) || (PieceColor.BLACK.equals(pieceColor) && numberOfBlackPawns < 7)
?